src/api/controller/user_controller.py

```diff
@@ -1,13 +1,27 @@
-from fastapi import APIRouter, HTTPException
-from service.user_service import UserService
-from models.user import User
+from fastapi import APIRouter, HTTPException, Depends
+from src.api.service.user_service import UserService
+from src.api.models.user import User
 
 router = APIRouter()
 user_service = UserService()
 
-@router.get("/user/{user_id}", response_model=User)
+@router.get("/user/{user_id}", response_model=User, responses={
+    400: {"description": "Bad request, invalid user ID"},
+    401: {"description": "Unauthorized access"},
+    404: {"description": "User not found"},
+})
 async def get_user(user_id: int):
+    if user_id <= 0:
+        raise HTTPException(status_code=400, detail="Bad request, invalid user ID")
+    
+    if not is_authorized():  
+        raise HTTPException(status_code=401, detail="Unauthorized access")
+
```

Comment:
We should do two type of validation, one is request body or parameter validation, in this case it checks if the user_id is a valid long/integer.

Authorization is a separate module, it is a good thing you considered this, but I planned to have separate library to perform authorization. This would automatically validate all request to reduce this check on every endpoint in every service.